### ISSUE TITLE
Attribute – Aufenthalte

### ISSUE THREAD

**Issue by connexia-age** [connexiadev/Vodamep#18]
*Regel*
Alle Aufenthalte mit den Aufnahmedaten bis zur ersten Lücke müssen in der entsprechenden Jahresmeldung vorhanden sein.
Warum gilt das nicht für die Attribute? 
Fehler: "Attribute, die nicht im Meldungszeitsaum liegen sind nicht erlaubt."

**Grund 1**
Erleichterung: Attribute müssen nicht umständlich auf den 1.1. des Meldungszeitraums gelegt werden.

**Grund 2**
Vollständiges Bild: Wenn wir in der Jahresmeldung ein vollständiges Bild über den Aufenthalt einer Person haben möchten (Stay, Admission, evtl. Leaving), warum zählen Attribute nicht dazu?

<img width="390" alt="image" src="https://user-images.githubusercontent.com/39065468/151043743-54ab22b2-74c7-41f5-a170-0265ef0b7841.png">

**Comment by dgorbach** [comment]
Ich sehe das so:
In einem Datenpaket sollen nur Daten enthalten sein, die für die Validierung oder die Auswertung des Meldezeitraums relevant sind.

ad Grund 1: Eine zeitliche Begrenzung der Daten macht dafür die Datenhaltung und Validierung einfacher. Wie ginge man damit um, wenn sich Daten der Attribute von vor x Jahren ändern oder nicht nicht mehr gemeldet werden? Z.B. weil ein Softwarewechsel stattfindet. 

ad Grund 2: Das vollständige Bild soll nur den Meldezeitraum betreffen. Attribute außerhalb des Meldezeitraums sind nicht relevant und sollten deshalb nicht Teil des Datenpaketes sein.

Aufenthalte auch außerhalb des Meldezeitraums, weil: 
- Für die Validierung der Dauer von Übergangspflege und Urlaub notwendig.
- Für die Auswertung der Gesamtdauer eines Aufenthalts notwendig.

Aufnahmedaten:
- Für die Auswertung notwendig: vorheriger Wohnort, Grund der Aufnahme, etc.

**Comment by connexia-age** [comment]
Wenn die Auswertbarkeit der Daten eine Rolle spielt, müsste ja gewährleistet werden, dass Wechsel von Finanzierung und Pflegestufen bei der Auswertung genau keine Rolle spielen.
Denn dafür müsste man Meldungen zusammenbauen.

Seht ihr das anders?

**Comment by dgorbach** [comment]
Versteh ich nicht. 
Für Auswertungen vom Jahr 2022 spielen doch nur die Pflegestufen aus dem Jahr 2022 eine Rolle. Zu jedem Zeitpunkt - im Jahr 2022 - eines Aufenthaltes im einer Person muss die Pflegestufe im Datenpaket definiert sein.  Gab es Wechsel, sind die entsprechenden Werte mit von und bis abgegrenzt alle in der Meldung.

Wozu muss man da dann Meldungen zusammenbauen?

**Comment by connexia-age** [comment]
Die Argumentation der Auswertungen passt aus meiner Sicht nicht: 

Hier wird der gesamte Aufenthalt für Auswertungen als Argument herangezogen: "Für die Auswertung der Gesamtdauer eines Aufenthalts notwendig."

Hier wird argumentiert, dass nur 2022 Daten relevant sind: "Für Auswertungen vom Jahr 2022 spielen doch nur die Pflegestufen aus dem Jahr 2022 eine Rolle."

Möchte ich jetzt also die Anzahl Wechsel der Pflegestufen, während des gesamten Aufenthalts, müssen Meldungen betrachtet werden. Oder verstehe ich das falsch?

Ich bin der Meinung, wir bauen eine Schnittstelle bei der die führende Entität der aktuelle Aufenthalt der Person ist. Und der soll vollständig sein.

**Comment by dgorbach** [comment]
Ich wusste nicht, dass die Anzahl der Wechsel der Pflegestufen ausgewertet werden soll. Wenn die Pflegestufen über den Meldezeitraum hinaus ausgewertet werden sollen, dann müsste man das nochmal überdenken.

Gibt es eine Beschreibung aller gewünschten Auswertungen?

**Comment by connexia-age** [comment]
Nein. Aber es gibt ja auch keine Anforderung für die Auswertung der Gesamtdauer eines Aufenthalts. 

Nicht falsch verstehen: 

- Für mich stellt der aktuelle Ansatz keine große Herausforderung dar und es ist beim Export der Bestandsdaten schon so umgesetzt.
- Ich werte auch keine Daten aus, kann das gerne aber weiterleiten. Ich sehe da nur eine gewisse Inkonsistenz. 
- Wir können das auch so belassen, bis wir eine Antwort haben.

**Comment by dgorbach** [comment]
Dann wär ich dafür, dass wir das so belassen.

**Comment by connexia-age** [comment]
<img width="1137" alt="image" src="https://user-images.githubusercontent.com/39065468/156044749-c817538c-9a68-43ef-95af-12fd77b98310.png">

**Comment by dgorbach** [comment]
<img width="1309" alt="image" src="https://user-images.githubusercontent.com/5755074/156130903-47e5efc9-b584-4869-a1aa-d0b0f8dd62ff.png">

**Comment by connexia-age** [comment]
Auch das haben wir intern besprochen und sprechen es gerne mit euch noch durch. 
Wir sehen diese Werte als Aufenthaltsdaten. 

[Kommentar ](https://github.com/connexiadev/Vodamep/issues/22#issuecomment-1055690401) bei  #22

**Comment by connexia-age** [comment]
Nach gemeinsamer Abstimmung: 

Wird so belassen wie es im Moment ist. 

- Attributänderungen sind nur gültig, wenn deren Gültigkeits-Datum sich im Meldungszeitraum befindet
- Aufenthalte werden nach wie vor bis zur ersten Lücke rückwirkend übertragen
